File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsJava/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicsjava/domain/repository/AppRepository.java
package pl.edu.uwr.pum.dagger_hilt_room_basicsjava.domain.repository;

import androidx.lifecycle.LiveData;

import java.util.List;

import pl.edu.uwr.pum.dagger_hilt_room_basicsjava.data.Student;

public interface AppRepository {
    LiveData<List<Student>> readAllData();
    void insert(Student student);
}

File: 2022-2023/sourcecode/lab/lab13/DaggerHiltROOMBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/dagger_hilt_room_basicskotlin/domain/repository/AppRepository.kt
package pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.domain.repository

import androidx.lifecycle.LiveData
import pl.edu.uwr.pum.dagger_hilt_room_basicskotlin.data.Student

interface AppRepository {
    val readAllData: LiveData<List<Student>>
    suspend fun insert(student: Student)
}